frontend/src/utils.ts

```diff
@@ -34,6 +34,7 @@ export type VmStatus =
   | '' //the environment phase is unknown.
   | 'Importing' //the image of the environment is being imported.
   | 'Starting' //the environment is starting.
+  | 'ResourceQuotaExceeded' //the environment could not start because the resource quota is exceeded.
```

Comment:
I would completely remove the VmStatus type, since the new enum has been defined. By removing this type, you'll find out all the other occurrences.